File: julia/euler/002.jl
function fib(a::Int, b::Int, accumulator::Int)::Int
  c::Int = a + b
  if c >= 4e6 return accumulator end
  accumulator += c%2==0 ? c : 0
  return fib(b,c,accumulator)
end

fib(a::Int, b::Int) = fib(a, b, 0)
fib() = fib(1, 1)

File: python/euler/002.py
def fib(a = 1, b = 1, accumulator = 0):
    c = a + b
    if c >= 4e6: return accumulator
    accumulator += c if c%2==0 else 0
    return fib(b, c, accumulator)

if __name__ == "__main__":
    A = fib()
    print(f"fib() = {A}")
